hongit-api/src/main/kotlin/com/devfloor/hongit/api/article/application/ArticleService.kt

```diff
@@ -111,18 +154,47 @@ class ArticleService(
     }
 
     @Transactional(readOnly = true)
-    fun showAllByNickname(nickname: String): List<ArticleFeedResponse> {
-        val articles = userRepository.findByNicknameOrNull(nickname)
-            ?.let(articleRepository::findAllByAuthor)
+    fun showAllByNickname(nickname: String, page: Int, pageSize: Int): List<ArticleFeedResponse> {
+        val user: User = userRepository.findByNicknameOrNull(nickname)
             ?: EntityNotFoundException.notExistsNickname(User::class, nickname)
 
-        return articles
-            .filter { !it.anonymous }
+        val pageRequest: PageRequest = PageRequest.of(page, pageSize)
+        val articles = user
+            .let { articleRepository.findAllByAuthor(it, pageRequest) }
+
+        return articles.content.filter { !it.anonymous }
+            .map {
+                ArticleFeedResponse(
+                    article = it,
+                    articleOptions = articleOptionRepository.findAllByArticle(it),
+                    articleFavorites = articleFavoriteRepository.findAllByArticle(it),
+                    page = page,
+                    totalArticleCount = articleRepository.countAllByAuthor(user)
+                )
+            }
+    }
+
+    @Transactional(readOnly = true)
+    fun showAllByFavoritedUserId(userId: Long, page: Int, pageSize: Int): List<ArticleFeedResponse> {
+        val user: User = userRepository.findByIdOrNull(userId)
+            ?: EntityNotFoundException.notExistsId(User::class, userId)
+        val articleFavorites: List<ArticleFavorite> = articleFavoriteRepository.findAllByUser(user)
+        val pageRequest: PageRequest = PageRequest.of(page, pageSize)
+        val articles = articleFavorites
+            .let {
+                articleRepository.findAllByIdIn(it.map { it.id }, pageRequest)
+            }
```

Comment:
```suggestion
        val articles = articleFavorites
            .let {
                articleRepository.findAllByIdIn(it.map { it.id }, PageRequest.of(page, pageSize))
            }
```

한 군데서만 쓰인다면 그냥 인라인처럼 바로 넣어도 괜찮은거 같어

Reply:
반영완료